concepts/decorators/about.md

```diff
@@ -1,2 +1,210 @@
+They also return one thing - the function which will be run by the program.
+This is usually either the function they were added to or a slightly modified version of that function, but it does not have to be - it can be any function whatsoever.
+
+The simplest decorator - one that does nothing - can be written as follows:
+```py
+def do_nothing(function):
+    return function
+```
+```py
+def function4a():
+    return 4
+
+@do_nothing
+def function4b():
+    return 4
+```
+```py
+>>>function4a()
+4
+>>>function4b()
+4
+```
+As you can see, both functions do the same thing, and when the decorator is added nothing changes.
+
+It is common to use the short form, `f`, instead of `function` which we will be using in the rest of the examples.
+
+A decorator that registers a function could be written as follows:
+```py
+functions = []
+
+def register_function(f):
+    functions.append(f)
+    return f
+```
+```py
+>>>functions
+[]
+```
+```py
+@register_function
+def function5():
+    return 5
+```
+```py
+>>>functions
+[<function function5 at 0x000001CA840AA700>]
+>>>function5()
+5
+```
+As you can see, adding the decorator to the function adds the function into the list.
+It is important to note that this happens when the function is declared, not when it is called.
+
+Many decorators you write will change what is run when the function is called.
+These will need to return a different function to the one passed.
+This different function is written inside the decorator's function.
+
+This decorator doubles the result of the function:
+```py
+def double(f):
+    def wrapper():
+        func_result = f()
+        doubled = func_result * 2
+        return doubled
+    return wrapper
+```
+```py
+def function6a():
+    return 6
+
+@double
+def function6b():
+    return 6
+```
+```py
+>>>function6a()
+6
+>>>function6b()
+12
+```
+As you can see the decorator modifies the result of the function.
+
+Now lets take a look at what is actually happening in this decorator.
+
+On the first line, we have the normal function definition.
+But, when we go inside this function and take a look at the next line, we get another function definition.
+This function is what we will return at the end of the decorator.
+This means we are not returning the original function.
+Because of this, Python will not actually call the original function when a call is made to what looks like that function.
+Instead, Python will call the function we created, `wrapper()`.
+
+If we look inside our `wrapper()` function, we first see a call to the original function.
+This call will do exactly the same as if we had called the original function without the decoration to the function.
+Then we get a line that modifies this result.
+Finally we return the modified result, which is what we get when calling the function.
+
+If we apply this decorator to a function which takes a single number and triples it, then we should get a function that sextuples our number.
+Lets see what happens:

```

Comment:
This is a good step-by-step, but it is _a lot_ to follow in one, big chunk.  I'd suggest pulling out comments like 

>This call will do exactly the same as if we had called the original function without the decoration to the function.

and using them as comments _within_ code examples, so that students see them as they are looking at the code, and then putting things like

>Finally we return the modified result, which is what we get when calling the function. 

Above or below the code example.